### PR TITLE
JobBatches Clean-up

### PR DESCRIPTION
**Pull request by conckja1**
https://opuscapita.atlassian.net/browse/IPA-1669

# Problem / Feature
Job batches are piling up, and need to be cleaned up regularly

# Changes
- creating route and underlying business logic for job-batch cleanup
- designed as a "module" so using it in IPA services is as simple as possible

NOTE: unit tests run manually since we dont have yet build automation set up

### CODE REVIEW

File: src/db/storage/sequelizeStorage/storage.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const { getStorage: getJobStorage } = require('./Job');
  */
 const getStorage = db => {
   return {
+    db,
```
Comment (huukias) [inline src/db/storage/sequelizeStorage/storage.js:9]: Hmm sorry, I don't get it where this return value is needed

File: src/api/jobBatch/index.js [hunk 2]
```diff
@@ -231,6 +236,42 @@ class JobBatchApi {
...
+   */
+  async cleanUpJobBatches({
+    retentionPeriodDays,
+    cleanupBatchSize = CLEANUP_BATCH_SIZE,
+    maxCleanupRounds = MAX_CLEANUP_ROUNDS }
+  ) {
+    validateParameterExistence(retentionPeriodDays, 'retentionPeriodDays');
+    const untilDate = moment.utc().subtract(retentionPeriodDays, 'days').toDate();
+    let total = 0;
+    // Let's keep it in a limited for loop just in case.
```
Comment (huukias) [inline src/api/jobBatch/index.js:256]: In order to avoid mega calls to database or?

Comment (conckja1) [reply src/api/jobBatch/index.js:256]: yes, in production deleting 1k jobbatches can take even a minute

Comment (conckja1) [reply src/api/jobBatch/index.js:256]: not sure if it makes any difference to the DB performance wise but I thought hat having multiple smaller calls is better

File: src/db/storage/sequelizeStorage/storage.js (hunk 1 above)
Comment (conckja1) [reply src/db/storage/sequelizeStorage/storage.js:9]: here:       const api = new JobBatchApi(storage.db);
JobBatchApi constructor requires the db, not the storage.

Comment (conckja1) [reply src/db/storage/sequelizeStorage/storage.js:9]: I am open to improvement suggestions

File: src/api/jobBatch/index.js (hunk 2 above)
Comment (huukias) [reply src/api/jobBatch/index.js:256]: I am just maybe bit concerned if it would leave leftovers but then again, we can trigger this relatively often

Comment (conckja1) [reply src/api/jobBatch/index.js:256]: Exactly, we can run it even every day. Besides, all the needed parameters are configurable

**Review by huukias — APPROVED** [review]

**Pull request merged**